### ISSUE TITLE
Unable to train pettingzoo.classic.leduc/texas_holdem with rllib

### ISSUE THREAD

**Issue by MachengShen** [PettingZoo-Team/PettingZoo#337]
Hi, I saw the discussion in #211 w.r.t. the compatibility of holdem environment with rllib. However, i'm still not able to run the training with rllib, with pettingzoo==1.4.2, and rllib==1.2.0.

I got the following error from rllib: 

```
ray.exceptions.RayTaskError(IndexError): ray::RolloutWorker.par_iter_next() (pid=25399, ip=192.168.31.171)
  File "python/ray/_raylet.pyx", line 480, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 432, in ray._raylet.execute_task.function_executor
  File "/home/macheng/miniconda3/envs/implicitPBT_env/lib/python3.8/site-packages/ray/util/iter.py", line 1152, in par_iter_next
    return next(self.local_it)
  File "/home/macheng/miniconda3/envs/implicitPBT_env/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 327, in gen_rollouts
    yield self.sample()
  File "/home/macheng/miniconda3/envs/implicitPBT_env/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 662, in sample
    batches = [self.input_reader.next()]
  File "/home/macheng/miniconda3/envs/implicitPBT_env/lib/python3.8/site-packages/ray/rllib/evaluation/sampler.py", line 95, in next
    batches = [self.get_data()]
  File "/home/macheng/miniconda3/envs/implicitPBT_env/lib/python3.8/site-packages/ray/rllib/evaluation/sampler.py", line 224, in get_data
    item = next(self.rollout_provider)
  File "/home/macheng/miniconda3/envs/implicitPBT_env/lib/python3.8/site-packages/ray/rllib/evaluation/sampler.py", line 670, in _env_runner
    _process_policy_eval_results(
  File "/home/macheng/miniconda3/envs/implicitPBT_env/lib/python3.8/site-packages/ray/rllib/evaluation/sampler.py", line 1447, in _process_policy_eval_results
    env_id: int = eval_data[i].env_id
IndexError: list index out of range
``` 

Here is the script to reproduce the error:

```
from copy import deepcopy
from numpy import float32
import os
from supersuit import dtype_v0

import ray
from ray.rllib.agents.registry import get_agent_class
from ray.rllib.env import PettingZooEnv
from pettingzoo.classic import leduc_holdem_v1

from ray.tune.registry import register_env

if __name__ == "__main__":
    alg_name = "PPO"

    # function that outputs the environment you wish to register.
    def env_creator():
        env = leduc_holdem_v1.env()
        env = dtype_v0(env, dtype=float32)
        return env

    num_cpus = 1

    config = deepcopy(get_agent_class(alg_name)._default_config)

    register_env("leduc_holdem",
                 lambda config: PettingZooEnv(env_creator()))

    test_env = PettingZooEnv(env_creator())
    obs_space = test_env.observation_space
    act_space = test_env.action_space

    config["multiagent"] = {
        "policies": {
            "player_0": (None, obs_space, act_space, {}),
            "player_1": (None, obs_space, act_space, {}),
        },
        "policy_mapping_fn": lambda agent_id: agent_id
    }

    config["num_gpus"] = int(os.environ.get("RLLIB_NUM_GPUS", "0"))
    config["log_level"] = "DEBUG"
    config["num_workers"] = 1
    config["rollout_fragment_length"] = 30
    config["train_batch_size"] = 200
    config["horizon"] = 200
    config["no_done_at_end"] = False

    ray.init(num_cpus=num_cpus + 1)
    trainer = get_agent_class(alg_name)(env="leduc_holdem", config=config)

    trainer.train()
```
I'm not sure whether this problem is on the rllib side or pettingzoo side. Do i have to make sure that illegal actions are never taken? I appreciate if you could share some insight on how to fix this problem. Thanks!

**Comment by justinkterry** [comment]
@weepingwillowben

**Comment by weepingwillowben** [comment]
I don't understand the error message so I will have to run it myself to debug it, and our hardware has issues right now. So it might be a bit before I can deal with this problem.

**Comment by justinkterry** [comment]
So I took a brief look.

-You should upgrade to the newest version of PettingZoo for good measure
-To see where the error lives, try your code on different PettingZoo environments, but I'm almost certain the problem is in RLlib given that error.

We'll take a look into this, but as Ben said it may take a little bit. If this is super urgent you may want to try some basic debugging yourself.

**Comment by MachengShen** [comment]
Thanks for the comments. I have also tried some other classic environment (rps, connect_four) with pettingzoo version 1.4.2, and they work with rllib. But all the 'holdem' environments fail with the mentioned error message, so I thought maybe there is some assumptions made about how to train with the 'holdem' envs which is violated in rllib. It would be good if you have any clue what could be the difference between 'holdem' envs with rps/connect_four that cause this problem.

**Comment by justinkterry** [comment]
-Can you try with the newest version of PettingZoo for good measure? There's been various fixes
-Do Hanabi and Mahjong work?

**Comment by MachengShen** [comment]
I also tried with pettingzoo==1.6.0, supersuit==2.6.0, rlcard==0.2.8, ray[rllib]==1.2.0, but got the same error message.

I tried Hanabi and Mahjong, and they also led to the same error message.

**Comment by weepingwillowben** [comment]
This is a bug in RLlib that was fixed by this PR: https://github.com/ray-project/ray/pull/14031.

This fix was not released in ray 1.2.0, so you will need to install the nightly build.

**Comment by MachengShen** [comment]
ok, thanks for pointing me to this pr, i will look into it.
